tools/tool-testing/galaxy-utils.js

```diff
@@ -0,0 +1,251 @@
+var _ = require('underscore');
+var process = require('process');
+var selftest = require('../tool-testing/selftest.js');
+var Run = selftest.Run;
+var testUtils = require('../tool-testing/test-utils.js');
+var files = require('../fs/files.js');
+var utils = require('../utils/utils.js');
+var authClient = require('../meteor-services/auth-client.js');
+var auth = require('../meteor-services/auth.js');
+
+// Run curl with the given specifications. Return an instance of Run.
+var runCurl = function (/*args*/) {
```

Comment:
You know, I'd say "do your HTTP fetches in Node"... but actually this is pretty clever, since it lets you use the selftest timeouts.